tslint.json

```diff
@@ -1,12 +1,74 @@
 {
-  "extends": "tslint:recommended",
+  "extends": ["tslint-microsoft-contrib", "tslint:recommended"],
   "rules": {
     "array-type": [true, "generic"],
     "quotemark": [true, "single", "avoid-escape"],
     "variable-name": [true, "ban-keywords"],
     "no-empty-interface": false,
     "no-var-requires": false,
     "no-string-literal": false,
-    "semicolon": [true, "ignore-interfaces"]
+    "semicolon": [true, "always"],
+    "chai-prefer-contains-to-index-of": false,
+    "chai-vague-errors": false,
+    "export-name": true,
+    "function-name": false,
+    "import-name": true,
```

Comment:
Why should we enforce this? 
Example: consider following structure: list/Factory, item/Factory what if we want to import both modules?

Reply:
If we move to the new folder structure that @DarkoKukovec suggested:

```
components/
   | FactoryItem.tsx
   | FactoryItem.test.tsx
   | FactoryList.tsx
   | FactoryList.test.tsx
```

this won't be an issue anymore.

Reply:
Also - according to our existing rules you won't be able to do this since you have to export named(!):

```javascript
// file list/Factory.ts

export const Factory = 0;
```

and then you won't be able to import (or create for that matter) two modules with name `Factory` without importing them with `as` something else.